closure/goog/i18n/numberformat.js

```diff
@@ -624,22 +624,104 @@ goog.i18n.NumberFormat.prototype.subformatFixed_ =
     }
 
-    for (var i = 0; i < digitLen; i++) {
-      parts.push(String.fromCharCode(zeroCode + intPart.charAt(i) * 1));
+    // If there's more than 1 number grouping,
+    // figure out the length of the non-repeated groupings (on the right)
+    if (this.groupingArray_.length >= 2) {
+      for (var j = 1; j < this.groupingArray_.length; j++) {
+        nonRepeatedGroupCount += this.groupingArray_[j];
+      }
+    }
 
-      if (digitLen - i > 1 && this.groupingSize_ > 0 &&
-          ((digitLen - i) % this.groupingSize_ == 1)) {
-        parts.push(grouping);
+    // Anything left of the fixed number grouping is repeated,
+    // figure out the length of repeated groupings (on the left)
+    var repeatedDigitLen = digitLen - nonRepeatedGroupCount;
+    if (repeatedDigitLen > 0) {
+      // There are repeating digits and non-repeating digits
+      for (var i = 0; i < digitLen; i++) {
+        parts.push(String.fromCharCode(zeroCode + intPart.charAt(i) * 1));
+        if (digitLen - i > 1) {
+          currentGroupSize = this.groupingArray_[currentGroupSizeIndex];
+          if (i < repeatedDigitLen) {
+            // Process the left side (the repeated number groups)
+            var repeatedDigitIndex = repeatedDigitLen - i;
+            // Edge case if there's a number grouping asking for "1" group at
+            // a time; otherwise, if the remainder is 1, there's the separator
+            if (currentGroupSize === 1 ||
+                (currentGroupSize > 0 &&
+                (repeatedDigitIndex % currentGroupSize) === 1)
+            ) {
+              parts.push(grouping);
+            }
+          } else {
+            // Process the right side (the non-repeated fixed number groups)
+            if (currentGroupSizeIndex < this.groupingArray_.length) {
+              if (i === repeatedDigitLen) {
+                // Increase the group index because a separator
+                // has previously added in the earlier logic
+                currentGroupSizeIndex += 1;
+              } else {
+                // Otherwise, just iterate to the right side and
+                // add a separator once the length matches to the expected
+                if (currentGroupSize ===
+                    i - repeatedDigitLen - nonRepeatedGroupCompleteCount + 1) {
+                  parts.push(grouping);
+                  // Keep track of what has been completed on the right
+                  nonRepeatedGroupCompleteCount += currentGroupSize;
+                  currentGroupSizeIndex += 1; // Get to the next number grouping
+                }
+              }
+            }
+          }
+        }
       }
+    } else {
+      // There are no repeating digits and only non-repeating digits
+      var digitLenLeft = digitLen;
+      var rightToLeftParts = [];
+      // Start from the right most non-repeating group and work inwards
+      for (
+          currentGroupSizeIndex = this.groupingArray_.length - 1;
```

Comment:
Not a big fan of the opening paren by itself on a line. Move as much as you can to the opening line.